### ISSUE TITLE
What about tf.image.resize?

### ISSUE THREAD

**Issue by makercob** [adobe/antialiased-cnns#15]
@richzhang Hi. Thank you for work.
Since in TensorFlow 2.0, tf.image.resize supports gradients. Will it be more efficient or superior in performance to downsample with  aforementioned function(various interpolation methods are provided)?

**Comment by richzhang** [comment]
Hi, this codebase is in pytorch, but I guess the same question applies. Resizing with the convolution function allows us to control the exact kernel being used. Bilinear downsample function should be equivalent for Tri-3 kernel and average pool is equivalent for Rect-2. The downsample operation is relatively cheap compared to a normal convolution, so I didn't worry too much about computational efficiency. Further efficiency can be had by separating the convolution (do BlurPool in x-direction only, then in y-direction).

**Comment by mrgloom** [comment]
Seems older tensorflow bilinear resize function has aliasing problems:
https://github.com/trevor-m/tensorflow-bicubic-downsample

**Comment by richzhang** [comment]
That's unfortunate. I saw some issues with PyTorch's as well. It's best to implement your own downsample/upsampling to be sure
